src/wazuh_db/wdb_global.c

```diff
@@ -1325,7 +1325,7 @@ bool wdb_insert_dbsync(wdb_t * wdb, struct kv const *kv_value, char *data) {
         sqlite3_stmt *stmt = wdb_get_cache_stmt(wdb, query);
         
         if (NULL != stmt) {
-            char *field_value = strtok(data,"|");
+            char *field_value = strtok(data,"\0");
```

Comment:
Please, add a #define variable, i.e.: MSG_FIELDS_SEPARATOR, so that you can use it here instead of the real character. If in the future we realized we need to change this for something else, you'll just need to change it in one single place.